### PR TITLE
[KYUUBI #2207]Support newly added spark data types: TimestampNTZType

### PR DESCRIPTION
**Pull request by Nick-0723**
### _Why are the changes needed?_

support newly added data types: TimestampNTZType 
since  Spark3.3.0

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala [hunk 1]
```diff
@@ -39,6 +44,7 @@ object SchemaHelper {
     case _: DecimalType => TTypeId.DECIMAL_TYPE
     case DateType => TTypeId.DATE_TYPE
     case TimestampType => TTypeId.TIMESTAMP_TYPE
+    case ntz if ntz.getClass.getName.equals(TIMESTAMP_NTZ) => TTypeId.TIMESTAMP_TYPE
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:47]: why not use simple class name as intervals？

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala [hunk 2]
```diff
@@ -76,6 +77,10 @@ class SparkOperationSuite extends WithSparkSQLEngine with HiveMetadataTests with
       .add("c16", "binary", nullable = false, "16")
       .add("c17", "struct<X: string>", nullable = true, "17")
 
+    if (SPARK_ENGINE_MAJOR_MINOR_VERSION._1 >= 3 && SPARK_ENGINE_MAJOR_MINOR_VERSION._2 > 2) {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:80]: this does not work for 4.x, shall be (x>3 || (x==3 && y >=3)

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala (hunk 1 above)
Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:47]: I found intervals did not handle the getColumnSize ,and I'll fix it in another pr。 So I would replace the string with  constant variables ，but maybe simple class name is a good choice。

Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:47]: > why not use simple class name as intervals？

I found intervals did not handle the getColumnSize ,and I'll fix it in another pr。 So I would replace the string with constant variables ，but maybe simple class name is a good choice。

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala (hunk 2 above)
Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:80]: > 

ok

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala (hunk 1 above)
Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:47]: +1 for simple class name

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala (hunk 2 above)
Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:80]: Isn't is this : (major == 3 && minor >= 3)

Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala]: oops，I get you

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**